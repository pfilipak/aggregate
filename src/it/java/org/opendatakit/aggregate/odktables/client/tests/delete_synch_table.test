# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people
    name STRING false
    age INTEGER false
    weight DECIMAL false
    
setTablePermissions user1 user2 people true false false
    
    
# Test that user without delete permission can't delete table
cloneSynchronizedTable user2 people
deleteSynchronizedTable user2 people

setTablePermissions user1 user2 people true true false
deleteSynchronizedTable user2 people


# Test that owner can delete table
deleteSynchronizedTable user1 people


# Test that table is gone
removeTableSynchronization user2 people


# Clean up
deleteUser user1
deleteUser user2