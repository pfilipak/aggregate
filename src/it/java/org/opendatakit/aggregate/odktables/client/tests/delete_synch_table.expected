# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people
    
setTablePermissions user1 user2 people true false false
    
    
# Test that user without delete permission can't delete table
cloneSynchronizedTable user2 people
deleteSynchronizedTable user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.PermissionDeniedException: Requesting user does not have permission for that operation!

setTablePermissions user1 user2 people true true false
deleteSynchronizedTable user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.PermissionDeniedException: Requesting user does not have permission for that operation!


# Test that owner can delete table
deleteSynchronizedTable user1 people


# Test that table is gone
removeTableSynchronization user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.TableDoesNotExistException: Table with ID people does not exist!


# Clean up
deleteUser user1
deleteUser user2
